c/CMakeLists.txt

```diff
@@ -1,57 +1,8 @@
-cmake_minimum_required(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.2)
 project(libsbp)
 
-# Setup flags for Code Coverage build mode
-set(CMAKE_CXX_FLAGS_COVERAGE "${CMAKE_CXX_FLAGS_DEBUG} --coverage" CACHE STRING
-    "Flags used by the C++ compiler for building with code coverage."
-    FORCE )
-set(CMAKE_C_FLAGS_COVERAGE "${CMAKE_C_FLAGS_DEBUG} --coverage" CACHE STRING
-    "Flags used by the C compiler for building with code coverage."
-    FORCE )
-SET(CMAKE_EXE_LINKER_FLAGS_COVERAGE
-    "${CMAKE_EXE_LINKER_FLAGS_DEBUG} --coverage" CACHE STRING
-    "Flags used for linking binaries with code coverage."
-    FORCE )
-set(CMAKE_SHARED_LINKER_FLAGS_COVERAGE
-    "${CMAKE_SHARED_LINKER_FLAGS_DEBUG} --coverage" CACHE STRING
-    "Flags used by the shared libraries linker during builds with code coverage."
-    FORCE )
-mark_as_advanced(
-    CMAKE_CXX_FLAGS_COVERAGE
-    CMAKE_C_FLAGS_COVERAGE
-    CMAKE_EXE_LINKER_FLAGS_COVERAGE
-    CMAKE_SHARED_LINKER_FLAGS_COVERAGE )
-# Update the documentation string of CMAKE_BUILD_TYPE for GUIs
-set(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}" CACHE STRING
-  "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel Coverage."
-   FORCE )
-
-# Set project version using Git tag and hash.
-execute_process(
-  COMMAND git describe --dirty --tags --always
-  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-  RESULT_VARIABLE GIT_VERSION_FOUND
-  ERROR_QUIET
-  OUTPUT_VARIABLE GIT_VERSION
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-if (GIT_VERSION_FOUND)
-  set(VERSION "unknown")
-else (GIT_VERSION_FOUND)
-  set(VERSION ${GIT_VERSION})
```

Comment:
Do we not need this property anymore?  CMake docs specify that's it's used when generated the library? https://cmake.org/cmake/help/v3.7/prop_tgt/VERSION.html

Reply:
The original code wasn't ever setting the `VERSION` property, it was just making a variable called `VERSION`.

If we want to have that property set I can certainly add it in, but I'm not sure our way of getting the version number will work in most cases. It seems like CMake usually expects versions to be exactly in the format of `0.0.0.0` with no additional text before or after. Our git tags at the very least have a "v" character at the beginning.

Reply:
Ok, let's not worry about it for now.